tools/data-conversion/src/main.rs

```diff
@@ -549,3 +627,369 @@ fn binary_to_fixed_bit_slice(
+///
+/// # Returns
+///
+/// This function returns a `std::io::Result<()>` which is `Ok` if the operation
+/// is successful, or an `Err` if an I/O error occurs while writing to the file.
+fn intermediate_to_fixed(
+    inter_rep: IntermediateRepresentation,
+    filepath_send: &mut Option<File>,
+) -> io::Result<()> {
+    // Negate exp
+    let neg_exponent = -inter_rep.exponent;
+
+    // 10^-exp
+    let scale_factor = BigInt::from(10).pow(neg_exponent as u32);
+
+    // Convert mantissa to BigInt
+    let mantissa_bigint = BigInt::from(inter_rep.mantissa);
+
+    let mantissa_mult = mantissa_bigint * scale_factor;
+
+    // Apply the sign
+    let signed_value = if inter_rep.sign {
+        mantissa_mult
+    } else {
+        -mantissa_mult
+    };
+
+    // Handle placement of decimal point
+    let mantissa_str = signed_value.to_string();
+    let mantissa_len = mantissa_str.len();
+    let adjusted_exponent = inter_rep.exponent + mantissa_len as i32;
+
+    let string = if adjusted_exponent <= 0 {
+        // Handle case where the exponent indicates a number less than 1
+        let zero_padding = "0".repeat(-adjusted_exponent as usize);
+        format!("0.{}{}", zero_padding, mantissa_str)
+    } else if adjusted_exponent as usize >= mantissa_len {
+        // Handle case where the exponent is larger than the length of the mantissa
+        format!(
+            "{}{}",
+            mantissa_str,
+            "0".repeat(adjusted_exponent as usize - mantissa_len)
+        )
+    } else {
+        // Normal case
+        let integer_part = &mantissa_str[..adjusted_exponent as usize];
+        let fractional_part = &mantissa_str[adjusted_exponent as usize..];
+        format!("{}.{}", integer_part, fractional_part)
+    };
+
+    // Write the result to the file or stdout
+    if let Some(file) = filepath_send.as_mut() {
+        file.write_all(string.as_bytes())?;
+        file.write_all(b"\n")?;
+    } else {
+        io::stdout().write_all(string.as_bytes())?;
+        io::stdout().write_all(b"\n")?;
+    }
+
+    Ok(())
+}
+
+/// Converts a string representation of a hexadecimal number to the intermediate representation.
+///
+/// This function takes a string slice representing a hexadecimal number,
+/// converts it to a `BigUint`, determines the sign, and constructs an
+/// `IntermediateRepresentation` containing the sign, mantissa, and exponent.
+///
+/// # Arguments
+///
+/// * `hex_string` - A string slice containing the hexadecimal number to be converted.
+///
+/// # Returns
+///
+/// This function returns an `IntermediateRepresentation` containing the sign, mantissa,
+/// and exponent of the hexadecimal number.
+///
+/// # Panics
+///
+/// This function will panic if the input string cannot be parsed as a hexadecimal number.
+
+fn hex_to_intermediate(hex_string: &str) -> IntermediateRepresentation {
+    // Get sign value before converting string
+    let sign = !hex_string.starts_with('-');
```

Comment:
I'm not sure I understand why `binary_to_intermediate` assumes two's complement but `hex_to_intermediate` assumes that the sign is indicated with a `-` symbol. I guess I would expect these functions to be basically identical other than the radix used for parsing?